lisa/runners/lisa_runner.py

```diff
@@ -898,3 +900,21 @@ def _create_platform_requirement(self) -> Optional[schema.NodeSpace]:
             )
 
         return platform_requirement
+
+    def _capture_vm_log_in_test_dir(
+        self, environment: Environment, test_result: TestResult
+    ) -> None:
+        if not environment.source_test_result:
```

Comment:
Why does it need?

Reply:
environment.source_test_result is of type None| TestResult